src/datasets/containers.jl

```diff
@@ -46,4 +48,36 @@ isimagefile(file::String) = occursin(IMAGEFILE_REGEX, lowercase(file))
 const IMAGEFILE_REGEX = r"\.(gif|jpe?g|tiff?|png|webp|bmp)$"
 
 
-## TODO: TableDataset
+struct TableDataset{T}
+    table::T #Should implement Tables.jl interface
+    TableDataset{T}(table::T) where T = Tables.istable(table) ? new{T}(table) : error("Object doesn't implement Tables.jl interface")
+end
+
+TableDataset(path::String) = TableDataset(DataFrame(CSV.File(path)))
+TableDataset(table::T) where {T} = TableDataset{T}(table)
+
+function LearnBase.getobs(dataset::FastAI.Datasets.TableDataset{T}, idx) where {T}
+    if Tables.rowaccess(dataset.table)
+        for (index, row) in enumerate(Tables.rows(dataset.table))
+            if index==idx
+                return row
+            end
+        end
+    elseif Tables.columnaccess(dataset.table)
+        rowvals = []
+        for i in 1:(length ∘ Tables.columnnames)(dataset.table)
+            append!(rowvals, Tables.getcolumn(dataset.table, i)[idx])
```

Comment:
Should this be `push!` or `append!`?

Reply:
Yeah it should be `push!` only, considering single item is being added.